src/Network/NWConnectionGroup.cs

```diff
@@ -0,0 +1,276 @@
+		{
+			if (inboundMessage == null)
+				throw new ArgumentNullException (nameof (inboundMessage));
+			if (outboundMessage == null)
+				throw new ArgumentNullException (nameof (outboundMessage));
+
+			var contentPtr = (content == null) ? IntPtr.Zero : content.Handle;
+			nw_connection_group_reply (GetCheckedHandle (), inboundMessage.Handle, outboundMessage.Handle, contentPtr);
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_connection_group_send_message (OS_nw_connection_group group, /* [NullAllowed] DispatchData */ IntPtr content, /* [NullAllowed] */ OS_nw_endpoint endpoint, OS_nw_content_context context, BlockLiteral *handler);
+
+		delegate void nw_connection_group_send_completion_t (IntPtr block, IntPtr error);
+		static nw_connection_group_send_completion_t static_SendCompletion = TrampolineSendCompletion;
+
+		[MonoPInvokeCallback (typeof (nw_connection_group_send_completion_t))]
+		static void TrampolineSendCompletion (IntPtr block, IntPtr error)
+		{
+			var del = BlockLiteral.GetTarget<Action<NWError?>> (block);
+			if (del != null) {
+				var err = error == IntPtr.Zero ? null : new NWError (error, owns: false);
+				del (err);
+				err?.Dispose ();
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void Send (DispatchData? content, NWEndpoint? endpoint, NWContentContext context, Action<NWError?>? handler)
+		{
+			unsafe {
+				if (handler == null) {
+					nw_connection_group_send_message (GetCheckedHandle (),
+						content == null ? IntPtr.Zero : content.Handle,
+						endpoint == null ? IntPtr.Zero : endpoint.Handle,
+						context.Handle,
+						null);
+					return;
+				}
+
+				BlockLiteral block_handler = new BlockLiteral ();
+				block_handler.SetupBlockUnsafe (static_SendCompletion, handler);
+				try {
+					nw_connection_group_send_message (GetCheckedHandle (),
+						content == null ? IntPtr.Zero : content.Handle,
+						endpoint == null ? IntPtr.Zero : endpoint.Handle,
+						context.Handle,
+						&block_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_connection_group_set_receive_handler (OS_nw_connection_group group, uint maximum_message_size, bool reject_oversized_messages, BlockLiteral *handler);
+
+		delegate void nw_connection_group_receive_handler_t (IntPtr block, IntPtr content, IntPtr context, bool isCompleted);
+		static nw_connection_group_receive_handler_t static_ReceiveHandler = TrampolineReceiveHandler;
+
+		[MonoPInvokeCallback (typeof (nw_connection_group_receive_handler_t))]
+		static void TrampolineReceiveHandler (IntPtr block, IntPtr content, IntPtr context, bool isCompleted)
+		{
+			var del = BlockLiteral.GetTarget<NWConnectionGroupReceiveDelegate> (block);
+			if (del != null) {
+				var nsContent = new DispatchData (content, owns: false);
+				var nsContext = new NWContentContext (context, owns: false);
```

Comment:
This way you wouldn't need to manually Dispose later:

```suggestion
				using var nsContent = new DispatchData (content, owns: false);
				using var nsContext = new NWContentContext (context, owns: false);
```

Reply:
and they would still be disposed if `del` throws :)